jvm/src/test/scala/YamlTests.scala

```diff
@@ -0,0 +1,216 @@
+      val testsInfo = elemInfo.get("tests").asInstanceOf[ArrayList[Any]]
+      symbol -> getTestGroup(testsInfo)
+    }.toMap
+
+  /** Load a TestGroup from a part of the parsed YAML */
+  private def getTestGroup(testsInfo: ArrayList[Any]): TestGroup =
+    val tests = mutable.ArrayBuffer.empty[YamlTest]
+    val subgroups = mutable.Map.empty[String, TestGroup]
+    for possibleTestInfo <- testsInfo.asScala do
+      val testInfo = possibleTestInfo.asInstanceOf[JavaMap[Any, Any]]
+      if testInfo.containsKey("inputs") then
+        // A test that looks like
+        // - inputs: ["foo", "bar"]
+        //   outupt: "foobar"
+        val inputs = testInfo
+          .get("inputs")
+          .asInstanceOf[ArrayList[Any]]
+          .asScala
+          .toSeq
+          .map(javaToVyxal)
+        val code = Option(testInfo.get("code").asInstanceOf[String])
+        tests.append(YamlTest(inputs, code, getOutputCriteria(testInfo)))
+      else if testInfo.size() == 1 then
+        // Either a test group or a v2-style [...input]: output test
+        val firstEntry = testInfo
+          .entrySet()
+          .iterator()
+          .next()
+        firstEntry.getKey() match
+          case groupName: String =>
+            // It's a nested group of tests
+            subgroups.put(
+              groupName,
+              getTestGroup(firstEntry.getValue().asInstanceOf[ArrayList[Any]])
+            )
+          case inputs: ArrayList[?] =>
+            // It's a v2-style test
+            val output = javaToVyxal(firstEntry.getValue())
+            tests.append(
+              YamlTest(
+                inputs.asScala.toSeq.map(javaToVyxal),
+                None,
+                Seq(Criterion.Equals(output))
+              )
+            )
+        end match
+      else
+        throw new Error(
+          s"Invalid test format (no inputs given): $testInfo"
+        )
+      end if
+    end for
+
+    TestGroup(tests.toSeq, subgroups.toMap)
+  end getTestGroup
+
+  private def getOutputCriteria(testInfo: JavaMap[Any, Any]): Seq[Criterion] =
+    val criteria = mutable.ArrayBuffer.empty[Criterion]
+    val output = testInfo.get("output")
+    if output != null then criteria += Criterion.Equals(javaToVyxal(output))
+    val startsWith = testInfo.get("starts-with")
+    if startsWith != null then
+      criteria += Criterion.StartsWith(javaListToVyxal(startsWith))
+    val endsWith = testInfo.get("starts-with")
+    if endsWith != null then
+      criteria += Criterion.EndsWith(javaListToVyxal(endsWith))
+    val contains = testInfo.get("contains")
+    if contains != null then
+      criteria += Criterion.Contains(javaListToVyxal(contains))
+    val containsMonotonic = testInfo.get("contains-monotonic")
+    if containsMonotonic != null then
+      criteria +=
+        Criterion.Contains(javaListToVyxal(containsMonotonic), monotonic = true)
+
+    criteria.toSeq
+  end getOutputCriteria
+
+  private def javaToVyxal(obj: Any): VAny = obj match
```

Comment:
is this going to work with the native tests?

Reply:
No, Snakeyaml is a Java library. That's why I stuck it in the jvm/ folder instead of the shared/ folder

Reply:
We could use a multiplatform Scala library instead, but I think it's fine to run these tests only on the JVM. If an element does have potentially different behavior depending on the platform, we can stick into the `ElementTests` suite in the `shared/` folder

Reply:
I used circe here: https://github.com/Vyxal/VyxalS/pull/14

Would that work? Idk

Reply:
It almost certainly would, I was just too lazy to use something new and figured we wouldn't need to run the tests on other platforms. I'll move it to the shared folder and use circe-yaml later (or [snake-yaml](https://github.com/VirtusLab/scala-yaml) if you want)

Reply:
🤷  do what you want

Reply:
I don't see a problem with sticking with snake yaml

Reply:
Idc